applications/coreutils/tac.cpp

```diff
@@ -29,60 +30,44 @@ static CommandLine cmdline = CMDLINE(
     "Concatenate and print lines of file(s) in reverse.",
     "If no filename provided read from input stream");
 
-Result tac(const char *path)
+Result tac(Stream *const input_stream)
 {
-
-    __cleanup(stream_cleanup) Stream *stream = stream_open(path, OPEN_READ);
-
-    if (handle_has_error(stream))
-    {
-        return handle_get_error(stream);
-    }
-    FileState stat = {};
-    stream_stat(stream, &stat);
-
     size_t read;
     char buffer[1024];
     char sep = (separator == nullptr ? '\n' : separator[0]);
 
-    String s;
+    StringBuilder s, temp;
     Vector<String> lines(20);
 
-    while ((read = stream_read(stream, &buffer, 1024)) != 0)
+    while ((read = stream_read(input_stream, buffer, 1024)) != 0)
     {
-        if (handle_has_error(stream))
-        {
-            return handle_get_error(stream);
-        }
-        String tmp(buffer, read);
-        s += tmp;
+        s.append(String(buffer, read));
```

Comment:
s.append(buffer, read); No need to create a string object :)

Reply:
i couldnt find a method which takes in char* and size_t